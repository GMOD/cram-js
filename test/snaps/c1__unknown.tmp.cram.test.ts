import { testFile } from '../testUtil'
testFile('c1#unknown.tmp.cram')
