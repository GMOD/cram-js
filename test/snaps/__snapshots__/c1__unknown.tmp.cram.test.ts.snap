// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole c1#unknown.tmp.cram without error 1`] = `
[
  {
    "fileId": "-",
    "magic": "CRAM",
    "majorVersion": 3,
    "minorVersion": 0,
  },
  {
    "containerHeader": {
      "_endPosition": 45,
      "_size": 19,
      "alignmentSpan": 0,
      "crc32": 2905504709,
      "landmarks": [
        0,
        215,
      ],
      "length": 331,
      "numBases": 0,
      "numBlocks": 2,
      "numLandmarks": 2,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 45,
        "_size": 19,
        "alignmentSpan": 0,
        "crc32": 2905504709,
        "landmarks": [
          0,
          215,
        ],
        "length": 331,
        "numBases": 0,
        "numBlocks": 2,
        "numLandmarks": 2,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "c1",
              },
              {
                "tag": "LN",
                "value": "10",
              },
              {
                "tag": "M5",
                "value": "e224bd8a70e5466c6a033be886045c85",
              },
              {
                "tag": "UR",
                "value": "/Users/rbuels/dev/htslib/test/c1.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "Tests permuations of seq / qual being present or "*" in mapped",
                "value": "",
              },
            ],
            "tag": "CO",
          },
          {
            "data": [
              {
                "tag": "and unmapped forms.  Also tests MD/NM tag generation.",
                "value": "",
              },
            ],
            "tag": "CO",
          },
        ],
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "c1",
              },
              {
                "tag": "LN",
                "value": "10",
              },
              {
                "tag": "M5",
                "value": "e224bd8a70e5466c6a033be886045c85",
              },
              {
                "tag": "UR",
                "value": "/Users/rbuels/dev/htslib/test/c1.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "Tests permuations of seq / qual being present or "*" in mapped",
                "value": "",
              },
            ],
            "tag": "CO",
          },
          {
            "data": [
              {
                "tag": "and unmapped forms.  Also tests MD/NM tag generation.",
                "value": "",
              },
            ],
            "tag": "CO",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": true,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "BA": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 30,
          },
          "parametersBytes": 1,
        },
        "BB": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 42,
              },
              "parametersBytes": 1,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 37,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 15,
          },
          "parametersBytes": 1,
        },
        "BS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              2,
            ],
          },
          "parametersBytes": 4,
        },
        "CF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 16,
          },
          "parametersBytes": 1,
        },
        "DL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "FC": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 27,
          },
          "parametersBytes": 1,
        },
        "FN": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 26,
          },
          "parametersBytes": 1,
        },
        "FP": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 28,
          },
          "parametersBytes": 1,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 25,
          },
          "parametersBytes": 1,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "TS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {},
      "tagIdsDictionary": [
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 394,
      "_size": 18,
      "alignmentSpan": 10,
      "crc32": 1090398951,
      "landmarks": [
        199,
      ],
      "length": 593,
      "numBases": 50,
      "numBlocks": 12,
      "numLandmarks": 1,
      "numRecords": 6,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 1,
    },
    "data": [
      {
        "_endPosition": 394,
        "_size": 18,
        "alignmentSpan": 10,
        "crc32": 1090398951,
        "landmarks": [
          199,
        ],
        "length": 593,
        "numBases": 50,
        "numBlocks": 12,
        "numLandmarks": 1,
        "numRecords": 6,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 1,
      },
      {
        "features": [
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              40,
              40,
              40,
              40,
              40,
              40,
              40,
              40,
              40,
              40,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "i",
                "data": "C",
                "pos": 3,
                "refPos": 3,
              },
              {
                "code": "X",
                "data": 2,
                "pos": 6,
                "refPos": 5,
              },
              {
                "code": "D",
                "data": 1,
                "pos": 8,
                "refPos": 7,
              },
            ],
            "readGroupId": -1,
            "readLength": 10,
            "readName": "_sqm",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 599,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "i",
                "data": "C",
                "pos": 3,
                "refPos": 3,
              },
              {
                "code": "X",
                "data": 2,
                "pos": 6,
                "refPos": 5,
              },
              {
                "code": "D",
                "data": 1,
                "pos": 8,
                "refPos": 7,
              },
            ],
            "readGroupId": -1,
            "readLength": 10,
            "readName": "_sm",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 600,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "i",
                "data": "N",
                "pos": 3,
                "refPos": 3,
              },
              {
                "code": "D",
                "data": 1,
                "pos": 8,
                "refPos": 7,
              },
            ],
            "readGroupId": -1,
            "readLength": 10,
            "readName": "_m",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 601,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": [
              40,
              40,
              40,
              40,
              40,
              40,
              40,
              40,
              40,
              40,
            ],
            "readBases": "AACCCTCGTT",
            "readGroupId": -1,
            "readLength": 10,
            "readName": "_squ",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 602,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": "AACCCTCGTT",
            "readGroupId": -1,
            "readLength": 10,
            "readName": "_su",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 603,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 11,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": null,
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 0,
            "readName": "_u",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 604,
          },
        ],
        "header": {
          "_endPosition": 639,
          "_size": 41,
          "compressedSize": 37,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            1,
            10,
            6,
            0,
            10,
            9,
            11,
            12,
            15,
            16,
            25,
            26,
            27,
            28,
            30,
            255,
            255,
            255,
            255,
            15,
            226,
            36,
            189,
            138,
            112,
            229,
            70,
            108,
            106,
            3,
            59,
            232,
            134,
            4,
            92,
            133,
          ],
          "contentId": 0,
          "contentPosition": 598,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 528585955,
          "parsedContent": {
            "_endPosition": 431,
            "_size": 37,
            "contentIds": [
              11,
              12,
              15,
              16,
              25,
              26,
              27,
              28,
              30,
            ],
            "md5": [
              226,
              36,
              189,
              138,
              112,
              229,
              70,
              108,
              106,
              3,
              59,
              232,
              134,
              4,
              92,
              133,
            ],
            "numBlocks": 10,
            "numContentIds": 9,
            "numRecords": 6,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 10,
            "refSeqStart": 1,
          },
          "uncompressedSize": 37,
        },
      },
    ],
  },
]
`;
